pkg/ensurance/analyzer/analyzer.go

```diff
@@ -187,332 +187,161 @@ func (s *AnormalyAnalyzer) Analyze(state map[string][]common.TimeSeries) {
-						basicThrottleQosPriority = beInfluencedPodPriority
-					}
-				} else {
-					basicThrottleQosPriority = executor.ClassAndPriority{PodQOSClass: v1.PodQOSBestEffort, PriorityClassValue: 0}
-				}
-
-				allPods, err := s.podLister.List(labels.Everything())
-				if err != nil {
-					klog.Errorf("Failed to list all pods: %v", err)
-					continue
-				}
-
-				for _, pod := range allPods {
-
-					var qosPriority = executor.ClassAndPriority{PodQOSClass: pod.Status.QOSClass, PriorityClassValue: utils.GetInt32withDefault(pod.Spec.Priority, 0)}
-					if !qosPriority.Greater(basicThrottleQosPriority) {
-						var throttlePod executor.ThrottlePod
-						throttlePod.PodTypes = types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}
-						throttlePod.CPUThrottle.MinCPURatio = uint64(action.Spec.Throttle.CPUThrottle.MinCPURatio)
-						throttlePod.CPUThrottle.StepCPURatio = uint64(action.Spec.Throttle.CPUThrottle.StepCPURatio)
-
-						throttlePod.PodCPUUsage, throttlePod.ContainerCPUUsages = s.getPodUsage(string(stypes.MetricNameContainerCpuTotalUsage), stateMap, pod)
-						throttlePod.PodCPUShare, throttlePod.ContainerCPUShares = s.getPodUsage(string(stypes.MetricNameContainerCpuLimit), stateMap, pod)
-						throttlePod.PodCPUQuota, throttlePod.ContainerCPUQuotas = s.getPodUsage(string(stypes.MetricNameContainerCpuQuota), stateMap, pod)
-						throttlePod.PodCPUPeriod, throttlePod.ContainerCPUPeriods = s.getPodUsage(string(stypes.MetricNameContainerCpuPeriod), stateMap, pod)
-						throttlePod.PodQOSPriority = qosPriority
-						throttlePods = append(throttlePods, throttlePod)
-					}
-				}
-			}
+		action, ok := avoidanceMaps[dc.ActionName]
+		if !ok {
+			klog.Warningf("The action %s not found.", dc.ActionName)
+			continue
 		}
-	}
 
-	// combine the replicated pod
-	for _, t := range throttlePods {
-		if i := ae.ThrottleExecutor.ThrottleDownPods.Find(t.PodTypes); i == -1 {
-			ae.ThrottleExecutor.ThrottleDownPods = append(ae.ThrottleExecutor.ThrottleDownPods, t)
-		} else {
-			if t.CPUThrottle.MinCPURatio > ae.ThrottleExecutor.ThrottleDownPods[i].CPUThrottle.MinCPURatio {
-				ae.ThrottleExecutor.ThrottleDownPods[i].CPUThrottle.MinCPURatio = t.CPUThrottle.MinCPURatio
-			}
+		//step3 do Throttle merge FilterAndSortThrottlePods
+		if action.Spec.Throttle != nil {
+			throttlePods, throttleUpPods := s.GetThrottleActionPods(enableSchedule, dc, action, stateMap)
+			// combine the replicated pod
+			ae.ThrottleExecutor.Deduplicate(throttlePods.(executor.ThrottlePods), throttleUpPods.(executor.ThrottlePods))
+		}
 
-			if t.CPUThrottle.StepCPURatio > ae.ThrottleExecutor.ThrottleDownPods[i].CPUThrottle.StepCPURatio {
-				ae.ThrottleExecutor.ThrottleDownPods[i].CPUThrottle.StepCPURatio = t.CPUThrottle.StepCPURatio
-			}
+		//step4 do Evict merge FilterAndSortEvictPods
+		if action.Spec.Eviction != nil {
+			evictPods, _ := s.GetEvictActionPods(dc.Triggered, action)
+			// remove the replicated pod
```

Comment:
here is evict merge?

Reply:
fix the comment to
"get and deduplicate evictPods"